src/validation/rules/package_name.rs

```diff
@@ -18,33 +18,27 @@ use super::*;
 #[derive(Debug, Clone)]
 pub struct PackageName {
     /// Package name to enforce.
-    name: String,
+    name: crate::package::PackageName,
 }
 
 impl PackageName {
     /// Create new checker for this rule.
-    pub fn new(name: &str) -> Self {
-        Self { name: name.into() }
+    pub fn new(name: crate::package::PackageName) -> Self {
+        Self { name }
     }
 }
 
-fn is_prefix(prefix: &str, package: &str) -> bool {
-    prefix
-        .replace('-', "_")
-        .split('.')
-        .zip(package.split('.'))
-        .all(|(a, b)| a == b)
-}
-
 impl Rule for PackageName {
     fn rule_info(&self) -> &'static str {
         "Make sure that the protobuf package name matches the buffer package name."
     }
 
     fn check_package(&mut self, package: &Package) -> Violations {
-        if !is_prefix(&self.name, &package.name) {
+        let transposed = self.name.to_string().replace('-', "_");
```

Comment:
A few notes here:

- We do not need to do the replacement of `-` to `_` here, because the `is_prefix` does that for us
- I have named the `is_prefix` rather poorly because the name does not reflect what it does
- We are kind of violating some abstractions here, I think ideally `crate::package::PackageName` should have a `.transposed()` method that is the canonical way of converting buffrs package names to protocol buffer package names, and we should not hardcode it here (that is on me because I wrote this code :smile:)

Feel free to close this but I wanted to note it down and maybe get back to it at some point.

Reply:
We need it for logging

Reply:
I agree, I just don't have the time right now